config/config.go

```diff
@@ -1,266 +1,162 @@
 	// type could be bigquery/postgres/gcs
-	Type string `yaml:"type" koanf:"type"`
+	Type string `yaml:"type" mapstructure:"type" koanf:"type"`
 
 	// directory to find specifications
-	Path string `yaml:"path" koanf:"path"`
+	Path string `yaml:"path" mapstructure:"path" koanf:"path"`
 
 	// backup configuration
-	Backup map[string]string `yaml:"backup" koanf:"backup"`
+	Backup map[string]string `yaml:"path" mapstructure:"backup" koanf:"backup"`
 }
 
 type Job struct {
 	// directory to find specifications
-	Path string `yaml:"path"`
+	Path string `yaml:"path" mapstructure:"path"`
 }
 
-type ProjectConfig struct {
-	// per project
-	Global map[string]string `yaml:"global"`
+type Project struct {
+	Name   string            `yaml:"name" mapstructure:"name"`
+	Config map[string]string `yaml:"config" mapstructure:"config"`
+}
 
-	// per namespace
-	Local map[string]string `yaml:"local"`
+type Namespace struct {
+	Name   string            `yaml:"name" mapstructure:"name"`
+	Config map[string]string `yaml:"config" mapstructure:"config"`
 }
 
 type LogConfig struct {
 	// log level - debug, info, warning, error, fatal
-	Level string `yaml:"level"`
+	Level string `yaml:"level" mapstructure:"level"`
 
 	// format strategy - plain, json
-	Format string `yaml:"format"`
+	Format string `yaml:"format" mapstructure:"format"`
 }
 
 type ServerConfig struct {
 	// port to listen on
-	Port int `yaml:"port"`
+	Port int `yaml:"port" mapstructure:"port" default:"9100"`
 	// the network interface to listen on
-	Host string `yaml:"host"`
+	Host string `yaml:"host" mapstructure:"host" default:"0.0.0.0"`
 
 	// service ingress host for jobs to communicate back to optimus
-	IngressHost string `yaml:"ingress_host"`
+	IngressHost string `yaml:"ingress_host" mapstructure:"ingress_host"`
 
 	// random 32 character hash used for encrypting secrets
-	AppKey string `yaml:"app_key"`
+	AppKey string `yaml:"app_key" mapstructure:"app_key"`
 
-	DB                      DBConfig       `yaml:"db"`
-	Metadata                MetadataConfig `yaml:"metadata"`
-	ReplayNumWorkers        int            `yaml:"replay_num_workers"`
-	ReplayWorkerTimeoutSecs time.Duration  `yaml:"replay_worker_timeout_secs"`
-	ReplayRunTimeoutSecs    time.Duration  `yaml:"replay_run_timeout_secs"`
+	DB                      DBConfig       `yaml:"db" mapstructure:"db"`
+	Metadata                MetadataConfig `yaml:"metadata" mapstructure:"metadata"`
+	ReplayNumWorkers        int            `yaml:"replay_num_workers" mapstructure:"replay_num_workers" default:"1"`
+	ReplayWorkerTimeoutSecs time.Duration  `yaml:"replay_worker_timeout_secs" mapstructure:"replay_worker_timeout_secs" default:"120s"`
+	ReplayRunTimeoutSecs    time.Duration  `yaml:"replay_run_timeout_secs" mapstructure:"replay_run_timeout_secs"`
 }
 
 type DBConfig struct {
 	// database connection string
 	// e.g.: postgres://user:password@host:123/database?sslmode=disable
-	DSN string `yaml:"host"`
+	DSN string `yaml:"host" mapstructure:"dsn"`
```

Comment:
looks like using Salt, `yaml` is not necessary to be defined. lets try to remove it

Reply:
tried this and works fine